### PR TITLE
Refactor resources to align with VCD and TM tests

### PR DESCRIPTION
**Pull request by Didainius**
* Revalidates that acceptance tests can be run on VCD 
* Renames `vcd_vcenter` resource and data source to `vcd_tm_vcenter`. Returns the classic data source `vcd_vcenter` that was available before.
* Renames `vcd_nsxt_manager` -> `vcd_tm_nsxt_manager` as it is only available in TM. Returns the classic data source `vcd_nsxt_manager` that was available before.
* Adds `make testacc-coverage` command to run acceptance tests with coverage report
* Adjusts `TestAccDataSourceNotFound` to avoid picking Tenant Manager tests (prefix `vcd_tm*`)

### CODE REVIEW

**Comment by lvirbalas** [comment]
> Renames vcd_nsxt_manager -> vcd_tm_nsxt_manager as it is only available in TM.

If this PR is renaming resources, let's give the PR a more serious title. Right now it looks as if it's only changing tests:
"Align VCD and TM **tests**"

File: vcd/datasource_vcd_tm_nsxt_manager.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/go-vcloud-director/v3/govcd"
+	"github.com/vmware/go-vcloud-director/v3/types/v56"
+)
+
+// TODO: TM: validate compatibility with old data source
```
Comment (adambarreiro) [inline vcd/datasource_vcd_tm_nsxt_manager.go:13]: Not needed anymore, or do we plan to revisit again in future?

File: website/docs/d/tm_nsxt_manager.html.markdown [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+sidebar_current: "docs-vcd-data-source-tm-nsxt-manager"
+description: |-
+  Provides a data source for available Tenant Manager NSX-T manager.
+---
+
+# vcd\_tm\_nsxt\_manager
+
+Provides a data source for available Tenant Manager NSX-T manager.
+
+Supported in provider *v3.0+*
```
Comment (adambarreiro) [inline website/docs/d/tm_nsxt_manager.html.markdown:13]:
```suggestion
Supported in provider *v4.0+*
```

**Review by adambarreiro — COMMENTED** [review]
First scan

File: website/docs/d/tm_nsxt_manager.html.markdown [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+~> **Note:** This resource uses new VMware Cloud Director
+[OpenAPI](https://code.vmware.com/docs/11982/getting-started-with-vmware-cloud-director-openapi) and
+requires at least VCD *10.1.1+* and NSX-T *3.0+*.
```
[the remark this answers is not in the document]
Comment (Didainius) [reply website/docs/d/tm_nsxt_manager.html.markdown:17]: Good point. Not needed. Removed this at all. We only care about TM now and the title itself has `vcd_tm`. I simplified it.

File: vcd/datasource_vcd_tm_nsxt_manager.go (hunk 1 above)
Comment (Didainius) [reply vcd/datasource_vcd_tm_nsxt_manager.go:13]: Good point. Not needed. Removed this

File: website/docs/d/tm_nsxt_manager.html.markdown (hunk 2 above)
Comment (Didainius) [reply website/docs/d/tm_nsxt_manager.html.markdown:13]: Removed this at all. It is not even here before v4

**Review by adambarreiro — APPROVED** [review]

File: website/docs/r/vcenter.html.markdown [hunk 4]
```diff
@@ -61,11 +61,14 @@ The following attributes are exported on this resource:
 * `is_connected` - Defines if the vCenter server is connected.
 * `mode` - One of `NONE`, `IAAS` (scoped to the provider), `SDDC` (scoped to tenants), `MIXED` (both
   uses are possible)
-* `listener_state` - `INITIAL`, `INVALID_SETTINGS`, `UNSUPPORTED`, `DISCONNECTED`, `CONNECTING`,
+* `connection_status` - `INITIAL`, `INVALID_SETTINGS`, `UNSUPPORTED`, `DISCONNECTED`, `CONNECTING`,
   `CONNECTED_SYNCING`, `CONNECTED`, `STOP_REQ`, `STOP_AND_PURGE_REQ`, `STOP_ACK`
 * `cluster_health_status` - Cluster health status. One of `GRAY` , `RED` , `YELLOW` , `GREEN`
 * `version` - vCenter version
 * `uuid` - UUID of vCenter
+* `vcenter_host` - Host of Vcenter server
```
Comment (lvirbalas) [inline website/docs/r/vcenter.html.markdown:69]:
```suggestion
* `vcenter_host` - Host of vCenter server
```

Comment (Didainius) [reply website/docs/r/vcenter.html.markdown]: Fixed.

**Review by lvirbalas — APPROVED** [review]

**Pull request merged**